bson/unmarshal_test.go

```diff
@@ -278,3 +278,158 @@ func TestUnmarshalExtJSONWithUndefinedField(t *testing.T) {
 		})
 	}
 }
+
+func TestUnmarshalBSONWithUndefinedField(t *testing.T) {
+	// When unmarshalling BSON, fields that are undefined in the destination struct are skipped.
+	// This process must not skip other, defined fields and must not raise errors.
+	type expectedResponse struct {
+		DefinedField string `bson:"DefinedField"`
+	}
+
+	unmarshalExpectedResponse := func(t *testing.T, BSON []byte) *expectedResponse {
+		t.Helper()
+		responseDoc := expectedResponse{}
+		err := Unmarshal(BSON, &responseDoc)
+		assert.Nil(t, err, "Error unmarshalling BSON: %v", err)
```

Comment:
nit: error messages and test failures should start with a lowercase letter. We tend to make an exception for function names (e.g. `MarshalBSON error: %v"`), but that's not the case here.

Reply:
Cool, good to know for the future. Changed.